lib/haml/railtie.rb

```diff
@@ -1,22 +1,19 @@
-if defined?(ActiveSupport)
-  require 'haml/template/options'
-  ActiveSupport.on_load(:before_initialize) do
-    ActiveSupport.on_load(:action_view) do
-      require "haml/template"
-    end
-  end
-end
+require 'haml/template/options'
 
 module Haml
   class Railtie < ::Rails::Railtie
     initializer :haml do |app|
-      require "haml/template"
-      if defined?(::Sass::Rails::SassTemplate) && app.config.assets.enabled
-        require "haml/sass_rails_filter"
+
+      ActiveSupport.on_load(:action_view) do
```

Comment:
Here's a concrete example from inside rails. I think the initializer is required to ensure that as you say, `ActiveSupport` is available.

https://github.com/rails/rails/blob/fe1f4b2ad56f010a4e9b93d547d63a15953d9dc2/activejob/lib/active_job/railtie.rb